src/python/WMComponent/WorkQueueManager/WorkQueueManagerWorkPoller.py

```diff
@@ -55,9 +56,12 @@ def retrieveCondition(self):
         """
         _retrieveCondition_
         set true or false for given retrieve condion
-        i.e. thredshod on workqueue
+        TODO: only condition it checks now is schedd limit
+        But we can add other conditions.
+        i.e. user config for enabling schedd limit check or thredshod on workqueue, etc
         """
-        return True
+
+        return (not isJobLimitReachedInSchedd())
```

Comment:
How frequently is this called?  It does reload a few things on each invocation, meaning it would be fine to do it every few seconds - but not at 10Hz.